src/outputs/parthenon_hdf5.hpp

```diff
@@ -202,49 +210,58 @@ template <>
 void HDF5WriteAttribute(const std::string &name, const std::vector<bool> &values,
                         hid_t location);
 
-template <typename T>
-void HDF5WriteAttribute(const std::string &name, T value, hid_t location) {
+template <typename T, REQUIRES(implements<kokkos_view(T)>::value)>
+void HDF5WriteAttribute(const std::string &name, const T &view, hid_t location) {
+  PARTHENON_REQUIRE(view.span_is_contiguous(), "Only works for contiguous views");
+
+  // cpplint demands compile constants be all caps
+  constexpr size_t RANK = static_cast<size_t>(T::rank);
+  hsize_t dim[RANK];
+  for (size_t d = 0; d < RANK; ++d) {
+    dim[d] = view.extent_int(d);
+  }
+  const H5S data_space = H5S::FromHIDCheck(H5Screate_simple(RANK, dim, dim));
+  // works regardless of memory space of the view
+  auto *pdata = view.data();
+  auto view_h = Kokkos::create_mirror_view(view);
+  if constexpr (!std::is_same<typename T::memory_space, Kokkos::HostSpace>::value) {
+    Kokkos::deep_copy(view_h, view);
+    pdata = view_h.data();
+  }
+  auto type = getHDF5Type(pdata);
+  H5A const attribute = H5A::FromHIDCheck(
+      H5Acreate(location, name.c_str(), type, data_space, H5P_DEFAULT, H5P_DEFAULT));
+  PARTHENON_HDF5_CHECK(H5Awrite(attribute, type, pdata));
+}
+
+template <typename T, REQUIRES(implements<scalar(T)>::value)>
+void HDF5WriteAttribute(const std::string &name, const T &value, hid_t location) {
   std::vector<T> vec(1);
   vec[0] = value;
   HDF5WriteAttribute(name, vec, location);
 }
 
-template <typename T>
-std::vector<T> HDF5ReadAttributeVec(hid_t location, const std::string &name) {
-  std::vector<T> res;
-  auto type = getHDF5Type(res.data());
+template <typename D, typename S>
+void HDF5WriteAttribute(const std::string &name, const ParArrayGeneric<D, S> &view,
+                        hid_t location) {
+  return HDF5WriteAttribute(name, view.KokkosView(), location);
+}
```

Comment:
With a couple of small changes, we could probably make `ParArrayGeneric` conform to the `kokkos_view` concept and get rid of this overload. Probably not worth it though.